x/bank/internal/keeper/keeper.go

```diff
@@ -250,27 +249,23 @@ func (keeper BaseSendKeeper) SubtractCoins(ctx sdk.Context, addr sdk.AccAddress,
 		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, amt.String())
 	}
 
-	oldCoins, spendableCoins := sdk.NewCoins(), sdk.NewCoins()
+	resultCoins := sdk.NewCoins()
 
-	acc := keeper.ak.GetAccount(ctx, addr)
-	if acc != nil {
-		oldCoins = acc.GetCoins()
-		spendableCoins = acc.SpendableCoins(ctx.BlockHeader().Time)
-	}
+	unspendableCoins := keeper.GetUnspendableCoins(ctx, addr)
 
-	// For non-vesting accounts, spendable coins will simply be the original coins.
-	// So the check here is sufficient instead of subtracting from oldCoins.
-	_, hasNeg := spendableCoins.SafeSub(amt)
-	if hasNeg {
-		return amt, sdkerrors.Wrapf(
-			sdkerrors.ErrInsufficientFunds, "insufficient account funds; %s < %s", spendableCoins, amt,
-		)
+	for _, coin := range amt {
+		oldBalance := keeper.GetBalance(ctx, addr, coin.Denom)
+		newBalance := oldBalance.Sub(coin)
+		if newBalance.Amount.LT(unspendableCoins.AmountOf(coin.Denom)) {
```

Comment:
Why check unspendable coins instead of spendable coins? (like we previously had)

I suppose they are functionally equivalent, is this way easier?

Reply:
See above.